src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -410,6 +410,73 @@ public static string SelectProductNameForDirectory(Platform.XDG_Type dirpath)
         }
 #endif
 
+        /// <summary>
+        /// The code is copied from the .NET implementation.
+        /// </summary>
+        internal static string GetFolderPath(System.Environment.SpecialFolder folder)
+        {
+            return InternalGetFolderPath(folder);
+        }
+
+        /// <summary>
+        /// The API set 'api-ms-win-shell-shellfolders-l1-1-0.dll' was removed from NanoServer, so we cannot depend on 'SHGetFolderPathW'
+        /// to get the special folder paths. Instead, we need to rely on the basic environment variables to get the special folder paths.
+        /// </summary>
+        /// <returns>
+        /// The path to the specified system special folder, if that folder physically exists on your computer.
+        /// Otherwise, an empty string ("").
+        /// </returns>
+        private static string InternalGetFolderPath(System.Environment.SpecialFolder folder)
```

Comment:
This code is moved from `System.Management.Automation.Environment` to `SMA.Platform` because the former class is completely removed because all environment members we extended now come back in `System.Environment`. Maybe we need to open a bug to review all if/defs and Unix specific code?